src/fdcache_entity.cpp

```diff
@@ -2386,6 +2402,7 @@ int put_headers(const char* path, headers_t& meta, bool is_copy, bool use_st_siz
 
 int FdEntity::UploadPendingHasLock(int fd)
 {
+    const std::lock_guard<std::mutex> lock(fdent_data_lock);
```

Comment:
Reentrancy issues were found in FlushHasLock() and RowFlushHasLock(), so locking here is no longer possible.
More details are explained in the full comments of this PR.